### PR TITLE
Added Karma-Service. Fixes #17

### LINKED ISSUE #17: Implement karma system for bot
```
@<mention_name> ++
@<mention_name> --
```

Both the above should work and increase/decrease the value of score of the mentioned person. 

Note for anyone tackling this issue: The score should be persistent and must not be reset to zero on deployment (re-running of the node chatbot.js script)!

### PR DESCRIPTION
**Pull request by yashmehrotra**

### CODE REVIEW

**Comment by yashmehrotra** [comment]
``` sql
CREATE TABLE karma (username varchar(255), point smallint);
```

File: services/karmaService.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+const sqlite3 = require('sqlite3');
+const config = require('../config');
+const db = new sqlite3.Database(config.dbFilePath);
+
+function _getUserKarma(username, callback, message) {
```
Comment (526avijitgupta) [inline services/karmaService.js]: better to do:

```
const _getUserKarma = (username, callback, message) => {
```

File: services/karmaService.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  db.each("SELECT SUM(point) as totalKarma FROM karma WHERE username = ?", username,
+    function(err, row) {
```
Comment (526avijitgupta) [inline services/karmaService.js]: `(err, row) => {`

File: services/karmaService.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      return callback(message + row.totalKarma);
+  });
+}
+
+module.exports = {
+  getKarma: (callback) => {
```
Comment (526avijitgupta) [inline services/karmaService.js:25]: This doesn't seem to be used anywhere. If not, then it should be removed.

Comment (yashmehrotra) [reply services/karmaService.js:25]: This functionality is remaining, basically, it is to view the karma of everyone

Comment (526avijitgupta) [inline services/karmaService.js]: Where is this being used ?
And it doesn't seem to be doing anything apart from `true` ?

**Comment by 526avijitgupta** [comment]
@yashmehrotra Please update your branch with the latest version of master. Apparently the tests were running correctly on my machine but were never returning with a non-zero exit status on Travis due to some wierd issue. Fixed that just now.

File: services/karmaService.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    let userKarmaMapping = {};
+    db.all("SELECT SUM(point) as totalKarma, username FROM karma GROUP BY username ORDER BY totalKarma DESC",
+      (err, rows) => {
+        _parseAllUserKarma(callback, rows);
+    });
+  },
+  giveKarma: (callback, username, value) => {
+    if ([-1,1].indexOf(value) > -1) {
```
Comment (526avijitgupta) [inline services/karmaService.js]: Can also be done by:

```
if (~[-1, 1].indexOf(value)) {
```

Completely your choice though!

File: services/karmaService.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      // Insert into DB
+      db.serialize(function() {
+        db.run("INSERT INTO karma (username, point) VALUES($username, $point)", {
+          $username: username,
+          $point: value,
+        });
+        let message = `Karma of ${username} is: `;
```
Comment (526avijitgupta) [inline services/karmaService.js]: Perhaps use `const` instead of `let` ?

**Comment by 526avijitgupta** [comment]
Looks good. A few general questions though:
- Do you plan on writing tests for this ?
- Do you think that this should only work if the mention name starts with a "@" ? As in 526avijitgupta and @526avijitgupta should not be different!

**Comment by yashmehrotra** [comment]
@526avijitgupta
1. Not feeling like writing tests, will open a issue for that
2. Will add a feature that people who are part of the room can be karma'd, so only @526avijitgupta  will work. (Also to be done later)

**Comment by 526avijitgupta** [comment]
Sounds good. Merging this in now.

**Pull request merged**